sklearn/utils/_metadata_requests.py

```diff
@@ -1438,14 +1448,35 @@ def __init_subclass__(cls, **kwargs):
         ----------
         .. [1] https://www.python.org/dev/peps/pep-0487
         """
+
+        def is_RequestMethod(obj, name: str):
+            value = inspect.getattr_static(obj, name)
+            return isinstance(value, RequestMethod)
+
         try:
             for method in SIMPLE_METHODS:
+                set_method_name = f"set_{method}_request"
                 requests = cls._get_class_level_metadata_request_values(method)
-                if not requests:
+                if hasattr(cls, set_method_name) and not is_RequestMethod(
+                    cls, set_method_name
+                ):
+                    # The method is not a descriptor, which means it's explicitly\
+                    # defined in the class, therefore we skip overriding it with a
+                    # descriptor here.
+                    # This happens, for instance in Scorers where the
+                    # `set_score_request` method is defined in the class.
+                    continue
+
+                if not requests and not hasattr(cls, set_method_name):
+                    # It can be the case that requests here is empty due to removing
+                    # metadata with an UNUSED marker, and that the parent class has
+                    # the method already. In this case, we cannot remove the method
+                    # since it exists in the parent class, and instead we set it with
+                    # an empty set of requests.
```

Comment:
Aren't we simply not setting any RequestMethod for such cases instead of an empty set of requests?

Reply:
The issue is "not setting a RequestMethod" means we inherit the one from the parent class, which in the case of `LogisticRegressionCV` means having one accepting a `sample_weight`

Reply:
Got it.